src/regex/bytes.rs

```diff
@@ -1557,18 +1557,25 @@ impl<'h> core::fmt::Debug for Match<'h> {
     fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
         let mut fmt = f.debug_struct("Match");
         fmt.field("start", &self.start).field("end", &self.end);
-        if let Ok(s) = core::str::from_utf8(self.as_bytes()) {
-            fmt.field("bytes", &s);
+
+        let bytes = self.as_bytes();
+        let formatted = bytes_to_string_with_invalid_utf8_escaped(bytes);
```

Comment:
Can you use [`regex_automata::util::escape::DebugHaystack`](https://docs.rs/regex-automata/latest/regex_automata/util/escape/struct.DebugHaystack.html) instead? It will basically do what you have here, but will only escape invalid UTF-8. What you've implemented here will escape not only invalid UTF-8, but all UTF-8 that isn't ASCII. (I think that would be a cure worse than the disease.)

Reply:
Modified to use DebugHaystack. I thought there would be such a feature but couldn't find it. Thanks for your suggestion. https://github.com/rust-lang/regex/pull/1203/commits/88112b31c21fcd5cd6d57fdb6a056f0a96880166